### ISSUE TITLE
Remove `options.obj` from API

### ISSUE THREAD

**Issue by alshakero** [Palindrom/Palindrom#133]
Currently, we pass an object `obj` reference in `options` to the constructor. And we used that reference everywhere all the time, inside Palindrom and outside. Because all that `jsonpatch` needed to observe was an object reference. 

But now as we are moving towards proxies, they rendered that reference useless, because they dictate using the new Proxy object reference to be able to track its changes. And this meant everyone is now required to wait for Palindrom to return its proxy observed object in `callback` callback. Any changes before this point will go unnoticed. 

Also, Palindrom wasn't using any data inside the `options.obj`, it was used only for its reference. 

In conclusion, with proxies, there isn't any purpose to pass an initial object. Unless we utilise (deep clone) it as the initial state.

**Comment by alshakero** [comment]
Implimented in https://github.com/Palindrom/Palindrom/pull/135
